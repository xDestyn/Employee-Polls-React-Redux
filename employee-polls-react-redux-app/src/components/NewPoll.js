// @ts-nocheck
import React from 'react';
import { useState } from 'react';
import { connect } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { handleAddQuestion } from '../actions/questions';
import Authenticate from './Authenticate';

const NewPoll = ({ dispatch, authedUser }) => {
  console.log(`NewPoll - user: ${authedUser}`);
  const navigate = useNavigate();

  const [optionOne, setOptionOne] = useState('');
  const [optionTwo, setOptionTwo] = useState('');
  const [success, setSuccess] = useState(false);
  const [error, setError] = useState(false);

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log(`Handle Submit - optionOne ${optionOne}`);
    console.log(`Handle Submit - optionTwo ${optionTwo}`);

    const question = {
      optionOneText: optionOne,
      optionTwoText: optionTwo,
    };

    try {
      dispatch(handleAddQuestion(question));
      console.log(`Dispatch - handleAddQuestion - Success`);
      setSuccess(true);
      setError(false);
    } catch (e) {
      console.log(`Dispatch - handleAddQuestion - Error`);
      setSuccess(false);
      setError(true);
    }

    // dispatch(handleAddQuestion(question))
    //   .then((x) => {
    //     console.log(`X: ${x}`);
    //     if (!x) {
    //       console.log(`Before Success - ${success}`);
    //       console.log(`Before Error - ${error}`);
    //       console.log(`Setting success to false`);
    //       console.log(`Setting error to true`);
    //       setSuccess(false);
    //       setError(true);
    //       console.log(`After Success - ${success}`);
    //       console.log(`After Error - ${error}`);
    //     } else {
    //       console.log(`Setting success to true`);
    //       console.log(`Setting error to false`);
    //       console.log(`Before Success - ${success}`);
    //       console.log(`Before Error - ${error}`);
    //       setSuccess(true);
    //       setError(false);
    //       console.log(`After Success - ${success}`);
    //       console.log(`After Error - ${error}`);
    //     }
    //   })
    //   .catch(() => {
    //     console.log(`Dispatch - handleAddQuestion - Error`);
    //   });

    setOptionOne('');
    setOptionTwo('');
    navigate('/');
  };

  const handleChangeOptionOne = (e) => {
    console.log(`Option one submit: ${e.target.value}`);
    setOptionOne(e.target.value);
  };

  const handleChangeOptionTwo = (e) => {
    console.log(`Option two submit: ${e.target.value}`);
    setOptionTwo(e.target.value);
  };

  return (
    <div className="new-poll-div">
      <Authenticate />
      {success && (
        <h1 data-testid="success-header" className={'Success'}>
          Poll Submitted!
        </h1>
      )}
      {error && (
        <h1 data-testid="error-header" className={'Error'}>
          Please enter required properties
        </h1>
      )}
      <h1>Create your own poll!</h1>
      <h2>Would You Rather</h2>
      <form name="poll-form" className="new-poll">
        <label className="selection-one-label" htmlFor="optionOneValue">
          Option One
          <input
            data-testid="option-one-input"
            placeholder="Option One"
            onChange={handleChangeOptionOne}
            value={optionOne}
            name="optionOneValue"
          ></input>
        </label>
        <label className="selection-two-label" htmlFor="optionTwoValue">
          Option Two
          <input
            data-testid="option-two-input"
            placeholder="Option Two"
            onChange={handleChangeOptionTwo}
            value={optionTwo}
            name="optionTwoValue"
          ></input>
        </label>
        <button data-testid="submit-button" type="submit" onClick={handleSubmit}>
          Submit
        </button>
      </form>
    </div>
  );
};

const mapStateToProps = ({ authedUser }) => {
  return { authedUser };
};

export default connect(mapStateToProps)(NewPoll);
