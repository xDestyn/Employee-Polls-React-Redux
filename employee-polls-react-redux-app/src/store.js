import { configureStore } from '@reduxjs/toolkit';
import reducer from './reducers';
import logger from './middleware/logger';
import thunk from 'redux-thunk';

const store = configureStore({
  reducer,
  middleware: [thunk, logger],
});

export default store;
