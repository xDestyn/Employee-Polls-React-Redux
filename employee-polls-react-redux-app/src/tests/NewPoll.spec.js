/* eslint-disable testing-library/no-unnecessary-act */
// @ts-nocheck
/* eslint-disable testing-library/render-result-naming-convention */
import NewPoll from '../components/NewPoll';
import * as React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

let root;

beforeEach(() => {
  root = document.createElement('div');
  document.body.appendChild(root);
});

afterEach(() => {
  // cleanup on exiting
  ReactDOM.unmountComponentAtNode(root);
  root.remove();
  root = null;
});

describe('NewPoll', () => {
  it('should match snapshot', () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <NewPoll />
        </Provider>
      </MemoryRouter>,
      root,
    );
    expect(component).toMatchSnapshot();
  });

  it('should add a new poll', async () => {
    await act(async () =>
      render(
        <MemoryRouter>
          <Provider store={store}>
            <NewPoll />
          </Provider>
        </MemoryRouter>,
        root,
      ),
    );

    const optionOne = screen.getByTestId('option-one-input');
    const optionTwo = screen.getByTestId('option-two-input');
    const submitButton = screen.getByTestId('submit-button');

    // Fire events
    await act(async () => {
      fireEvent.change(optionOne, { target: { value: 'Workout' } });
      fireEvent.change(optionTwo, { target: { value: 'Play Video Games' } });
      fireEvent.click(submitButton);
    });

    expect(screen.getByTestId('success-header')).toBeInTheDocument();
    expect(screen.queryByTestId('error-header')).not.toBeInTheDocument();
  });

  it('should not add a new poll due to not receiving second option', async () => {
    await act(async () =>
      render(
        <MemoryRouter>
          <Provider store={store}>
            <NewPoll />
          </Provider>
        </MemoryRouter>,
        root,
      ),
    );

    const optionOne = screen.getByTestId('option-one-input');
    const optionTwo = screen.getByTestId('option-two-input');
    const submitButton = screen.getByTestId('submit-button');

    // Fire events
    await act(async () => {
      fireEvent.change(optionOne, { target: { value: undefined } });
      fireEvent.change(optionTwo, { target: { value: undefined } });
      fireEvent.click(submitButton);
    });

    expect(screen.getByTestId('error-header')).toBeInTheDocument();
    expect(screen.queryByTestId('success-header')).not.toBeInTheDocument();
  });
});
