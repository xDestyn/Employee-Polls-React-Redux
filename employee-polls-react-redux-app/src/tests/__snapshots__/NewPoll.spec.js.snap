// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="new-poll-div"
      >
        <h1>
          Create your own poll!
        </h1>
        <h2>
          Would You Rather
        </h2>
        <form
          class="new-poll"
          name="poll-form"
        >
          <label
            class="selection-one-label"
            for="optionOneValue"
          >
            Option One
            <input
              data-testid="option-one-input"
              name="optionOneValue"
              placeholder="Option One"
              value=""
            />
          </label>
          <label
            class="selection-two-label"
            for="optionTwoValue"
          >
            Option Two
            <input
              data-testid="option-two-input"
              name="optionTwoValue"
              placeholder="Option Two"
              value=""
            />
          </label>
          <button
            data-testid="submit-button"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="new-poll-div"
    >
      <h1>
        Create your own poll!
      </h1>
      <h2>
        Would You Rather
      </h2>
      <form
        class="new-poll"
        name="poll-form"
      >
        <label
          class="selection-one-label"
          for="optionOneValue"
        >
          Option One
          <input
            data-testid="option-one-input"
            name="optionOneValue"
            placeholder="Option One"
            value=""
          />
        </label>
        <label
          class="selection-two-label"
          for="optionTwoValue"
        >
          Option Two
          <input
            data-testid="option-two-input"
            name="optionTwoValue"
            placeholder="Option Two"
            value=""
          />
        </label>
        <button
          data-testid="submit-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
